eng/common/TestResources/New-TestResources.ps1

```diff
@@ -129,6 +128,40 @@ if (!$templateFiles) {
     exit
 }
 
+# If there is a value for $SubscriptionConfiguration look it up and set
+# script-level variables for subsequent steps.
+if ($PSBoundParameters.ContainsKey('SubscriptionConfiguration')) {
+    Write-Verbose "Using subscription configuration $SubscriptionConfiguration from KeyVault $KeyVaultName..."
+    $keyVaultSecret = ConvertTo-SecureString -String $KeyVaultAppSecret -AsPlainText -Force
+    $keyvaultCredential = [System.Management.Automation.PSCredential]::new($KeyVaultAppId, $keyVaultSecret)
+
+    $keyVaultAccount = Retry {
+        Connect-AzAccount -Tenant $KeyVaultTenantId -Credential $keyvaultCredential -ServicePrincipal -Environment $Environment
+    }
+
+    $exitActions += {
+        Write-Verbose "Logging out of service principal '$($keyVaultAccount.Context.Account)'"
+        $null = Disconnect-AzAccount -AzureContext $keyVaultAccount.Context
+    }
+
+    $subscriptionSecret = Get-AzKeyVaultSecret -VaultName $KeyVaultName -Name $SubscriptionConfiguration
+    $subscriptionParameters = ($subscriptionSecret.SecretValueText | ConvertFrom-JSON)
+
+    $SubscriptionId = $subscriptionParameters.SubscriptionId
+    $TenantId = $subscriptionParameters.TenantId
+    $TestApplicationId = $subscriptionParameters.TestApplicationId
+    $TestApplicationSecret = $subscriptionParameters.TestApplicationSecret
+    $TestApplicationOid = $subscriptionParameters.TestApplicationOid
+    $ProvisionerApplicationId = $subscriptionParameters.ProvisionerApplicationId
+    $ProvisionerApplicationSecret = $subscriptionParameters.ProvisionerApplicationSecret
+    $Environment = $subscriptionParameters.Environment
```

Comment:
The parameters that were deleted appear here now. This allows the rest of the script's logic to run without change, however there may be some strange corner cases involving `@PSBoundParameters` in scripts that are expecting parameters which were deleted.